department-of-veterans-affairs/caseflow
# Kev ma/appeals 38955

Resolves [38955](https://jira.devops.va.gov/browse/APPEALS-38955)

# Description
- Created `AutoAssignAlertBanner.jsx` and `BatchAutoAssignButton.jsx` components
- Added Jest tests: `AutoAssignAlertBanner.test.js` and `BatchAutoAssignButton.test.js`
- Updated ReviewPackage Redux Store to include autoAssign
- Added autoAssign reducers and actions to reviewPackage actions and reducers
- Updated `correspondence_controller#auto_assign_correspondences`
- Added `correspondence_controller#auto_assign_status`
- Added Rspec tests for auto_assign_status
- Linting and Code Climate fixes
- Added FeatureToggles to raise a banners for testing purposes
  - `auto_assign_banner_failure`
  - `auto_assign_banner_max_queue`
  - `auto_assign_banner_no_rpt`

## Acceptance Criteria

## Testing Plan

1. Sign-in as INBOUND OPS ADMIN user
2. Go to Queue > Inbound Ops Team team cases view
3. Click on Auto assign correspondences button
4. Green Banner appears when successful
5. Click on button again
6. Blue info/error banner appears as no correspondences are available for assignment
7. In Development/Demo, go to /test/users page either through the URL or by clicking on Switch Users in the user dropdown next to Search Cases in the header
8. Add "auto-assign-banner-failure" feature toggle to the list to manually trigger a Standard Error
9. Go back to Inbound Ops Team cases view and click on banner
10. Note that a red banner appears with UUID


# Frontend
## User Facing Changes

<img width="979" alt="image" src="https://github.com/department-of-veterans-affairs/caseflow/assets/104021955/ca91bae1-25d3-415a-a6c5-f4179823160f">

<img width="753" alt="image" src="https://github.com/department-of-veterans-affairs/caseflow/assets/104021955/f4419ead-4105-48b5-a14a-7d808119cbed">

<img width="1259" alt="image" src="https://github.com/department-of-veterans-affairs/caseflow/assets/104021955/5b95cd37-4d48-4963-bf07-15f7fdfd2ec6">


# Best practices
## Code Documentation Updates

## Tests
### Test Coverage
Did you include any test coverage for your code? Check below:

### Code Climate
Your code does not add any new code climate offenses? If so why?

## Monitoring, Logging, Auditing, Error, and Exception Handling Checklist

### Monitoring

### Logging

### Auditing

### Error Handling
